File: c++/prog12.cpp
#include <iostream>
#include <string>
using namespace std;

class student{
public:
char name[100];
int roll_no;
void setname();
void setroll_no();
};
void student :: setroll_no()
    {
    cout<<"enter the roll number of student : ";
    cin>>roll_no;
}
void student::setname(){
    {
    cout<<"enter the name : ";
    cin>>name;
}
}
class test : public student{
public: 
int marks1,marks2;
void setmarks();

};
void test :: setmarks(){
    cout<<"enter marks of subject1 and subject2 : "<<endl;
    cin>>marks1;
    cout<<"  ";
    cin>>marks2;
}

class result : public test{
public:
int result;
void setresult();
int getresult(void);
void display();
};
void result :: display(){
     cout<<"the name , roll no. , marks of subject1 and subject 2 , and their total is : "<<name<<","<<"roll no. is "<<roll_no<<" , mark of subject1 "<<marks1<<" , marks of subject2 is  "<<marks2<<" and their total is "<<(marks1+marks2)<<endl;

     cout<<"total is : "<<getresult();
    }
    int result :: getresult(){
        return (marks1+marks2);
    }
int main()
{
    result r1;
int size;
cout<<"enter the number of students : ";
cin>>size;
   for(int i=0;i<size;i++){
     r1.setname();
    r1.setroll_no();
    r1.setmarks();
r1.display();
   }
   
    return 0;
}

File: java/prog12.java

import java.util.Scanner;

class Student {

    String name;
    int roll_no;

    public void st_name(String name) {
        this.name = name;
    }

    public void roll_no(int roll_no) {
        this.roll_no = roll_no;
    }

    public String getname() {
        return name;
    }

    public int getroll() {
        return roll_no;
    }

}

class Test extends Student {

    int marks_maths;
    int marks_english;

    public void marks(int marks_english, int marks_maths) {
        this.marks_english = marks_english;
        this.marks_maths = marks_maths;
    }

    public void display() {
        System.out.println("the marks of maths is : " + marks_maths);
        System.out.println("the marks of marks_english is : " + marks_english);
    }

    public int engmarks() {
        return marks_english;
    }

    public int mathmarks() {
        return marks_maths;
    }

}

class Result extends Test {

    int total_score;

    @Override
    public void display() {
        total_score = marks_english + marks_maths;
        System.out.println("the name of student is " + getname() + " whose roll no. is " + getroll());
        System.out.println("the total score of both subjects is : " + total_score);
    }
}

public class prog12 {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        Result r = new Result();
        Test t = new Test();

        r.st_name("me");
        r.roll_no(1);
        r.marks(50, 90);
        r.display();

        sc.close();
    }
}
